### PR TITLE
Feature/8440 dsc list request

### PR DESCRIPTION
**Pull request by KaiTeuber**
## Description
Request to fetch DSCList from server

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-8440

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/CachingHTTPClient.swift [hunk 1]
```diff
@@ -189,6 +190,37 @@ class CachingHTTPClient: AppConfigurationFetching, StatisticsFetching, LocalStat
...
+					let DSCList = try SAP_Internal_Dgc_DscList(serializedData: package.bin)
+					let responseETag = response.httpResponse.value(forCaseInsensitiveHeaderField: "ETag")
+					let dscListResponse = DSCListResponse(dscList: DSCList, eTag: responseETag)
+					Log.info("Received DSCList \(try DSCList.jsonString())", log: .vaccination)
+					completion(.success(dscListResponse))
+				} catch {
+					completion(.failure(error))
+				}
+			case .failure(let error):
+				completion(.failure(error))
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Client/HTTP Client/CachingHTTPClient.swift:219]: Can we log the error here already? :)

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/CachingHTTPClient.swift [hunk 2]
```diff
@@ -189,6 +190,37 @@ class CachingHTTPClient: AppConfigurationFetching, StatisticsFetching, LocalStat
 		}
 	}
 
+	// MARK: Protocol DSCListFetching
+
+	func fetchDSCList(
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Client/HTTP Client/CachingHTTPClient.swift:195]:
```suggestion

	/// Fetches the DSC List
	/// - Parameters:
	///   - etag: an optional ETag to download only versions that differ the given tag
	///   - completion: result handler
	func fetchDSCList(
```

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift [hunk 3]
```diff
@@ -1492,7 +1492,38 @@ private extension URLRequest {
 	
 		return request
 	}
-	
+
+	static func DSCListRequest(
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift:1496]:
```suggestion
	static func requestDSCList(
```

File: src/xcode/ENA/ENA/Source/Client/__tests__/Mocks/CachingHTTPClientMock.swift [hunk 4]
```diff
@@ -127,5 +127,33 @@ final class CachingHTTPClientMock: CachingHTTPClient {
...
 	}
+
+	// MARK: Protocol DSCListFetching
+
+	static let staticDSCList: SAP_Internal_Dgc_DscList = {
+		guard
+			let url = Bundle(for: CachingHTTPClientMock.self).url(forResource: "default_dsc_list", withExtension: "bin"),
+			let data = try? Data(contentsOf: url),
+			let dscList = try? SAP_Internal_Dgc_DscList(serializedData: data)
+		else {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Client/__tests__/Mocks/CachingHTTPClientMock.swift:138]:
```suggestion
		else {
			Log.debug("Cannot initialize static test data", log: .vaccination)
			return SAP_Internal_Dgc_DscList()
		}
```

File: src/xcode/ENA/ENA/Source/Services/DSCList/DSCListProviding.swift [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import Foundation
+
+protocol DSCListFetching {
+	var configuration: HTTPClient.Configuration { get }
+	var session: URLSession { get }
+
+	typealias DSCListCompletionHandler = (Result<DSCListResponse, Error>) -> Void
+
+	func fetchDSCList(etag: String?, completion: @escaping DSCListCompletionHandler)
+
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/DSCList/DSCListProviding.swift:14]:
```suggestion
```

**Review by naveeddotio — APPROVED** [review]
suggested a few things

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/CachingHTTPClient.swift (hunk 2 above)
Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/Client/HTTP Client/CachingHTTPClient.swift:195]: done

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift (hunk 3 above)
Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift:1496]: I changed it to `dscListRequest` to keep is consistent. Request is always at the ende if the static func creating the requests.

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/CachingHTTPClient.swift [hunk 6]
```diff
@@ -189,6 +190,42 @@ class CachingHTTPClient: AppConfigurationFetching, StatisticsFetching, LocalStat
 		}
 	}
 
+	// MARK: Protocol DSCListFetching
+
+	/// Fetches lis of DSC certificates
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Client/HTTP Client/CachingHTTPClient.swift:195]:
```suggestion
	/// Fetches list of DSC certificates
```

**Review by ArturFriesen — APPROVED** [review]

**Pull request merged**